scripts/generate-CHANGES.py

```diff
@@ -0,0 +1,324 @@
+				# The preconditions have been met
+				# for word splitting to occur.
+				#
+				# Cut the word into two parts.
+				#
+				# The first part is the one that goes
+				# onto the empty line are are currently at,
+				# and fills it.
+				#
+				# The other part is the remainder. That part
+				# is put back into the word queue.
+				size = width - textwidth(wordPrefix)
+				keep = word[0:size]
+				putBack = word[size:]
+				queue.insert(0, putBack)
+				word = keep
+			else:
+				# There was not enough space left in the line to be
+				# able to contain this word.
+				#
+				# This is either because,
+				# (a) There simply was not enough space, or...
+				# (b) The word-splitting check demanded that
+				#     we skip the rest of the line. This
+				#     happens because we, as a rule, only
+				#     output split words beginning on their
+				#     own pristine line.
+				#
+				# Discard the word, and consider the current line complete.
+				# This moves on to the next line.
+				queue.insert(0, word)
+				lines.append(line)
+				line = ""
+				continue
+
+		# Add the word to the line
+		if line == indent:
+			line = wordPrefix + word
+		else:
+			line = wordPrefix + word
+
+	# If we have a non-empty line, but no
+	# words left in the queue, it is the last
+	# line. Append it to the 'lines' list.
+	if line != "":
+		lines.append(line)
+
+	# Format the lines for user consumption.
+	return "\n".join(lines)
+
+def matchRegexp(regexp, content):
+	'''
+		A simple function to make regular
+		expression matching return a boolean
+		value for easier readability.
+	'''
+	return re.match(regexp, content, re.UNICODE) is not None
+
+def gitLogOutput():
+	'''
+		Invoke git to get NUL-separated log lines.
+		The rows are as follows:
+		 - Commit hash
+		 - Author
+		 - Email
+		 - Date (YYYY-MM-DD)
+		 - Subject
+	'''
+	p = subprocess.Popen(["git", "log", "--date=short", "--pretty=format:%h%x00%an%x00%ae%x00%ad%x00%s"], stdout=subprocess.PIPE)
+	stdout, stderr = p.communicate()
+	if stdout is not None:
+		stdout = stdout.decode('utf-8')
+	if stderr is not None:
+		stderr = stderr.decode('utf-8')
+	if p.returncode != 0:
+		raise Exception('"git log" failed: %s', stderr)
+	return stdout
+
+def main():
+	log = gitLogOutput()
+
+	# The authors dict maps between email addresses
+	# and author names.
+	authors = {}
+
+	# The dates dictionary maps from
+	# dates, to another dictionary
+	# that maps from email addresses
+	# to a sorted list of commit subject
+	# lines.
+	#
+	# In pseudo-code, the type of 'dates' is:
+	# 
+	#     dict<string(YYMMDD), dict<string(email), list(subject)>>
+	#
+	dates = {}
+
+	# lastSubject contains the last subject line
+	# processed. It is used to detect duplicate
+	# subject lines and discard them.
+	lastSubject = ""
+
+	for line in log.split('\n'):
+		commit, author, email, date, subject = line.split('\0')
+
+		if author == "Thorvald Natvig":
+			# Because someone forgets to set their git username on every platform...
```

Comment:
...Or we could just accept that it rewrites the names. :)

I actually didn't put it in, because Git on my Mac doesn't support --mailmap. :(